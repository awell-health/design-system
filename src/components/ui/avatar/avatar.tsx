import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';

const avatarVariants = cva('', {
  variants: {
    variant: {
      circle: 'rounded-full',
      square: 'rounded-md'
    },
    size: {
      xs: 'w-8',
      sm: 'w-16',
      md: 'w-20',
      lg: 'w-32'
    }
  },
  defaultVariants: {
    variant: 'circle',
    size: 'sm'
  }
});

export interface AvatarProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof avatarVariants> {
  src?: string;
  placeholder?: boolean;
  text?: string;
}

function Avatar({
  src,
  className,
  variant,
  size,
  placeholder = false,
  text,
  ...props
}: AvatarProps) {
  return (
    <div className={cn('avatar', { placeholder: placeholder })}>
      {!placeholder && (
        <div className={cn(avatarVariants({ variant, size }), className)} {...props}>
          <img src={src} />
        </div>
      )}
      {placeholder && (
        <div
          className={cn(
            'bg-neutral text-neutral-content w-24 rounded-full',
            avatarVariants({ size }),
            className
          )}
          {...props}
        >
          <span
            className={cn({
              'text-xs': size === 'xs',
              'text-sm': size === 'sm',
              'text-xl': size === 'md',
              'text-3xl': size === 'lg'
            })}
          >
            {text}
          </span>
        </div>
      )}
    </div>
  );
}

export { Avatar, avatarVariants };
