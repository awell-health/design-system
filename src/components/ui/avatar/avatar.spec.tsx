import { describe, expect, it } from 'vitest';

import { Avatar } from './avatar';
import React from 'react';
import { render } from '@testing-library/react';

describe('Avatar', () => {
  it('match image snapshot', () => {
    const { container } = render(
      <Avatar src='https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"' />
    );

    expect(container).toMatchSnapshot();
  });

  it('match placeholder snapshot', () => {
    const { container } = render(<Avatar placeholder text='SV' />);

    expect(container).toMatchSnapshot();
  });
});
