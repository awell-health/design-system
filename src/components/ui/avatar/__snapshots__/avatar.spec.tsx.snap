// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > match image snapshot 1`] = `
<div>
  <div
    class="avatar"
  >
    <div
      class="rounded-full w-16"
    >
      <img
        src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80""
      />
    </div>
  </div>
</div>
`;

exports[`Avatar > match placeholder snapshot 1`] = `
<div>
  <div
    class="avatar placeholder"
  >
    <div
      class="bg-neutral text-neutral-content rounded-full w-16"
    >
      <span
        class=""
      >
        SV
      </span>
    </div>
  </div>
</div>
`;
